airbyte-workers/src/main/java/io/airbyte/workers/internal/DefaultStateAggregator.java

```diff
@@ -0,0 +1,47 @@
+package io.airbyte.workers.internal;
+
+import com.google.common.base.Preconditions;
+import io.airbyte.config.State;
+import io.airbyte.protocol.models.AirbyteStateMessage;
+import io.airbyte.protocol.models.AirbyteStateMessage.AirbyteStateType;
+
+public class DefaultStateAggregator implements StateAggregator {
+
+  private AirbyteStateType stateType = null;
+  private final StateAggregator streamStateAggregator = new StreamStateAggregator();
+  private final StateAggregator singleStateAggregator = new SingleStateAggregator();
+
+  @Override public void ingest(final AirbyteStateMessage stateMessage) {
+    checkTypeOrSetType(stateMessage.getType());
+
+    getStateAggregator().ingest(stateMessage);
+  }
+
+  @Override public State getAggregated() {
+    return getStateAggregator().getAggregated();
+  }
+
+  /**
+   * Return the state aggregator that match the state type.
+   */
+  private StateAggregator getStateAggregator() {
+    return switch (stateType) {
+      case STREAM -> streamStateAggregator;
+      case GLOBAL, LEGACY -> singleStateAggregator;
+    };
+  }
+
+  /**
+   * We can not have 2 different state types given to the same instance of this class.
+   * This method set the type if it is not. If the state type doesn't exist in the message, it is set to LEGACY
+   */
+  private void checkTypeOrSetType(AirbyteStateType stateType) {
+    if (stateType == null) {
+      stateType = AirbyteStateType.LEGACY;
```

Comment:
Will this actually set the type on the `AirbyteStateMessage` object in the method that calls this?

Reply:
Oh, I just misread this. I dont think you are actually trying to modify the `AirbyteStateMessage` object in the calling method, you are just trying to set this `stateType` variable so that it is populated for the check below

Reply:
This could be a bit more readable if the parameter in this method was called something like `inputStateType`

Reply:
Yeah without the coloration from IntelliJ it is harder to understand.